packages/plugin-core/src/ExtensionProvider.ts

```diff
@@ -24,6 +24,15 @@ export class ExtensionProvider {
     return ExtensionProvider.extension;
   }
 
+  /** Returns true if the extension is registered.
+   *
+   * This should be true most of the time, but you may need to check first in
+   * commands that run before or during initialization.
+   */
+  static isRegistered() {
```

Comment:
can we add this to [[Cook|dendron://dendron.docs/pkg.plugin-core.dev.cook]] and also note it in an async?

Reply:
Documented and created a PR for it. Will make an async once this is merged.

Reply:
I'm a bit weary of us using this in the wrong way - I think we may eventually get into initialization race conditions if we use this broadly and in the wrong way.  I also think of this ExtensionProvider static class as a temporary solution until we can have a better dependency injection model. 

I'm ok to leave it, but is it really necessary to have this per the other comment around accessing vs code settings w/o the Extension being activated?

Reply:
Makes sense. See my other comment, removing this interface. I instead moved the function to `VSCodeUtils`, and deprecated `ExtensionProvider.getWorkspaceConfig()` to avoid the dependency on the extension.